### ISSUE TITLE
TypeScript: regular promises not compatible with RSVP?

### ISSUE THREAD

**Issue by NullVoxPopuli** [emberjs/ember-test-waiters#129]
Tried upgrading to 2.0.0 again, but am getting these type errors:

```ts
app/services/toast.ts:28:20 - error TS2345: Argument of type 'Promise<unknown>' is not assignable to parameter of type 'import("C:/Users/Preston/Documents/Development/NullVoxPopuli/emberclear/packages/frontend/node_modules/@types/rsvp/index").Promise<unknown>'.
  Property 'new' is missing in type 'Promise<unknown>' but required in type 'import("C:/Users/Preston/Documents/Development/NullVoxPopuli/emberclear/packages/frontend/node_modules/@types/rsvp/index").Promise<unknown>'.

28     waitForPromise(createToast(status, message, options));
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/@types/rsvp/index.d.ts:103:9
    103         new<T>(
                ~~~
    'new' is declared here.
```

whether or not I have `@types/rsvp` installed myself seems to not matter.

I'm not sure why I'm getting an RSVP error now? unless ember-test-waiters uses TS and is using an old version of `@types/rsvp`? 🤔

**Comment by scalvert** [comment]
Thanks, @NullVoxPopuli. I could be wrong, but I think the issue here is you're trying to intermix native and RSVP promises.

https://github.com/NullVoxPopuli/emberclear/blob/master/packages/frontend/app/services/toast.ts#L57

You're not importing RSVP promise in this service, but rather using native promises, so `waitForPromise` will error due to a type mismatch.

**Comment by NullVoxPopuli** [comment]
> waitForPromise will error due to a type mismatch.

right, but waitForPromise is compat with native Promises, so the types should be updated, yeah?

https://github.com/emberjs/ember-test-waiters/pull/130

**Comment by scalvert** [comment]
We specifically [import `RSVP.Promise`](https://github.com/emberjs/ember-test-waiters/blob/master/addon/wait-for-promise.ts#L2) and use that in `waitForPromise`. So I'm not sure that's true?

**Comment by scalvert** [comment]
We additionally specifically use `RSVP.Promise` [in our tests](https://github.com/emberjs/ember-test-waiters/blob/master/tests/unit/wait-for-promise-test.ts#L6), so we're not testing for native promises.

**Comment by NullVoxPopuli** [comment]
sure, but isn't RSVP.Promise pretty much a polyfill at this point? the APIs are the same

**Comment by NullVoxPopuli** [comment]
like, should I not use `waitForPromise` / ember-test-waiters if I want to use native promises? seems weird to need to fork the repo for a type error :-\

**Comment by scalvert** [comment]
Looking at this again, and playing with it a while, I think we should stick with what you have. If we use `PromiseLike<T>`, that interface doesn't include `catch` and `finally`. That means that we'd be narrowing the return type from `waitForPromise`, which isn't what we want to be doing.

**Comment by scalvert** [comment]
Resolved with #130
